components/ProfilePage/EditAvatarDialog.js

```diff
@@ -0,0 +1,203 @@
+    }
+  }
+};
+
+const getInitialAvatarData = user => {
+  if (user.avatarData) {
+    if (typeof user.avatarData === 'string') return JSON.parse(user.avatarData);
+    if (typeof user.avatarData === 'object') return user.avatarData;
+  }
+  return generateRandomOpenPeepsAvatar();
+};
+
+function EditAvatarDialog({ user, onClose = () => {} }) {
+  const [updateUser, { loading }] = useMutation(UPDATE_USER, {
+    onCompleted() {
+      onClose();
+    },
+  });
+
+  const classes = useStyles();
+
+  const [avatarType, setAvatarType] = useState(user.avatarType || 'OpenPeeps');
+  const [avatarData, setAvatarData] = useState(getInitialAvatarData(user));
+
+  const handleSubmit = e => {
+    e.preventDefault();
+
+    const newAvatarData =
+      avatarType === 'OpenPeeps' ? JSON.stringify(avatarData) : user.avatarData;
+    const newAvatarUrl = getAvatarUrl(user, avatarType);
+    user.avatarType = avatarType;
+    user.avatarData = newAvatarData;
+    user.avatarUrl = newAvatarUrl;
+
+    updateUser({
+      variables: {
+        avatarType: avatarType,
+        avatarData: newAvatarData,
+      },
+    });
+  };
+
+  // TODO: is this the best way to maintain state of object?
+  const setAvatarField = (field, value) => {
+    setAvatarData({ ...avatarData, [field]: value });
+  };
+
+  return (
+    <Dialog onClose={onClose} open>
+      <DialogTitle>{t`Edit profile picture`}</DialogTitle>
+      <form onSubmit={handleSubmit}>
+        <DialogContent dividers>
+          <Tabs
+            value={avatarType}
+            variant="fullWidth"
+            textColor="primary"
+            scrollButtons="off"
+            onChange={(e, tab) => setAvatarType(tab)}
+          >
+            <Tab
+              classes={{ root: classes.tabRoot }}
+              value="OpenPeeps"
+              label="OpenPeeps"
+            />
+            <Tab
+              classes={{ root: classes.tabRoot }}
+              value="Gravatar"
+              label="Gravatar"
+              disabled={
+                !user.availableAvatarTypes ||
+                !user.availableAvatarTypes.some(val => val === 'Gravatar') ||
+                !user.email
+              }
+            />
+            <Tab
+              classes={{ root: classes.tabRoot }}
+              value="Facebook"
+              label="Facebook"
+              disabled={
+                !user.availableAvatarTypes ||
+                !user.availableAvatarTypes.some(val => val === 'Facebook') ||
+                !user.facebookId
+              }
+            />
+            <Tab
+              classes={{ root: classes.tabRoot }}
+              value="Github"
+              label="Github"
+              disabled={
+                !user.availableAvatarTypes ||
+                !user.availableAvatarTypes.some(val => val === 'Github') ||
+                !user.githubId
+              }
+            />
+          </Tabs>
+          <div className={classes.avatarEditor}>
```

Comment:
Seems that removing this div can make mobile look even better; buttons can wrap and all buttons become clickable
![image](https://user-images.githubusercontent.com/108608/109304728-a33c2000-7877-11eb-9ea6-e4eaa809b6bd.png)

And on desktop it looks decent as well.